### DISCUSSION TITLE
Text outline

### DISCUSSION THREAD

**Discussion by Nutri626** [Chatterino/chatterino2#4082] [Ideas]
hello, is there way to enable text outline of the chat, like in other programs (twitch downloader for example) there is a setting for that so on it if you have transparency on and chat is on top of white background text is still visible cos of outline but if you do the same with chatterino (basically white text on white background) text is barely visible, it would be really helpful if there was an option for that

**Comment by pajlada** [comment]
There is not functionality for that right now, no. I'm not opposed to different text rendering modes such as outlines as options.

**Reply by Nutri626** [reply]
is it hard stuff to do from devs perspective or not so? like i will donate and stuff just for that option to be available as fast as possible lol

**Reply by pajlada** [reply]
It's not trivial so it would require some research into it before we'd be able to figure out how complicated it is. The best thing to do in this situation is to write out as detailed of a feature request as you can to make it easier for people to figure out what type of feature it is you want exactly, what the use case is, how you envision it being used etc

**Reply by Nutri626** [reply]
i believe program that i referred to (twitch downloader, https://github.com/lay295/TwitchDownloader) is "open source" and should not be hard to look into how it is done to have text outline (that exact option)
![image](https://user-images.githubusercontent.com/41973452/233733792-8a655123-edb5-41b9-abda-eefeeb67f1a3.png)


as i explained above if you do transparent chatbox by yourself for viewing fullscreen stream with chatbox added to it (since there is no official transparency option for that even tho i saw people asking for it for years) since text is white and if there is white background you can barely see any text (example)
![image](https://user-images.githubusercontent.com/41973452/233733913-f5ba3000-007a-4b3d-a639-a22611868e99.png)


how it looks with outline enabled on white background, big difference 
![image](https://user-images.githubusercontent.com/41973452/233734013-d91ac536-1dcf-47e8-addd-0bb5adace1a3.png)

**Reply by LosFarmosCTL** [reply]
> i believe program that i referred to (twitch downloader, https://github.com/lay295/TwitchDownloader) is "open source" and should not be hard to look into how it is done to have text outline 

That program is written in an entirely different language + framework (C# and WPF compared to C++ and Qt), so just taking what they did for rendering will not be possible, a chatterino version would be completely different code.

**Reply by Nutri626** [reply]
> > i believe program that i referred to (twitch downloader, https://github.com/lay295/TwitchDownloader) is "open source" and should not be hard to look into how it is done to have text outline
> 
> That program is written in an entirely different language + framework (C# and WPF compared to C++ and Qt), so just taking what they did for rendering will not be possible, a chatterino version would be completely different code.

i did not mean just taking/copying it but at least it should be explained how its done and see how its possible in this (if its even possible at all), its just a font setting nothing special imho

**Reply by LosFarmosCTL** [reply]
> i did not mean just taking/copying it but at least it should be explained how its done and see how its possible in this (if its even possible at all), its just a font setting nothing special imho

Rendering text is completely different when comparing WPF to Qt, so no, there is literally nothing chatterino could take from the implementation of another program, except for the resulting look in the end.

**Reply by Nutri626** [reply]
> > i did not mean just taking/copying it but at least it should be explained how its done and see how its possible in this (if its even possible at all), its just a font setting nothing special imho
> 
> Rendering text is completely different when comparing WPF to Qt, so no, there is literally nothing chatterino could take from the implementation of another program, except for the resulting look in the end.

well shit ... got it
